OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -151,6 +168,9 @@ public void ResolveOrder(Actor self, Order order)
 			if (!order.Queued)
 				self.CancelActivity();
 
+			if (Info.DeployAsGroup && deployState == DeployState.Deployed && IsGroupDeployNeeded(self))
```

Comment:
Put this above the `!order.Queued` check - if you drop the order here, it already acted out as a Stop already.